All the MacOS tests fail with the same error:

```Python
self = <hyperspy.drawing.signal1d.Signal1DFigure object at 0x13bcbe780>
    def _draw_animated(self):
        if self.ax.figure:
            canvas = self.ax.figure.canvas
>           canvas.restore_region(self._background)
E           AttributeError: 'Signal1DFigure' object has no attribute '_background'

hyperspy/drawing/figure.py:28: AttributeError
```

So I'm guessing its related to that the MacOS environment doesn't create the canvas when its supposed to (i.e. when calling `s.plot()`)?

--- a/hyperspy/tests/io/test_hdf5.py
+++ b/hyperspy/tests/io/test_hdf5.py
@@ -371,6 +371,7 @@ class TestAxesConfiguration:
 
 class Test_permanent_markers_io:
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_save_permanent_marker(self):
         s = Signal2D(np.arange(100).reshape(10, 10))
         m = markers.point(x=5, y=5)
@@ -379,6 +380,7 @@ class Test_permanent_markers_io:
             filename = tmp + '/testsavefile.hdf5'
         s.save(filename)
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_save_load_empty_metadata_markers(self):
         s = Signal2D(np.arange(100).reshape(10, 10))
         m = markers.point(x=5, y=5)
@@ -391,6 +393,7 @@ class Test_permanent_markers_io:
         s1 = load(filename)
         assert len(s1.metadata.Markers) == 0
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_save_load_permanent_marker(self):
         x, y = 5, 2
         color = 'red'
@@ -412,6 +415,7 @@ class Test_permanent_markers_io:
         assert m1.marker_properties['color'] == color
         assert m1.name == name
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_save_load_permanent_marker_all_types(self):
         x1, y1, x2, y2 = 5, 2, 1, 8
         s = Signal2D(np.arange(100).reshape(10, 10))
@@ -442,6 +446,7 @@ class Test_permanent_markers_io:
         for m0_dict, m1_dict in zip(m0_dict_list, m1_dict_list):
             assert m0_dict == m1_dict
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_save_load_horizontal_line_marker(self):
         y = 8
         color = 'blue'
@@ -458,6 +463,7 @@ class Test_permanent_markers_io:
         m1 = s1.metadata.Markers.get_item(name)
         assert san_dict(m1._to_dictionary()) == san_dict(m._to_dictionary())
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_save_load_horizontal_line_segment_marker(self):
         x1, x2, y = 1, 5, 8
         color = 'red'
@@ -475,6 +481,7 @@ class Test_permanent_markers_io:
         m1 = s1.metadata.Markers.get_item(name)
         assert san_dict(m1._to_dictionary()) == san_dict(m._to_dictionary())
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_save_load_vertical_line_marker(self):
         x = 9
         color = 'black'
@@ -491,6 +498,7 @@ class Test_permanent_markers_io:
         m1 = s1.metadata.Markers.get_item(name)
         assert san_dict(m1._to_dictionary()) == san_dict(m._to_dictionary())
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_save_load_vertical_line_segment_marker(self):
         x, y1, y2 = 2, 1, 3
         color = 'white'
@@ -508,6 +516,7 @@ class Test_permanent_markers_io:
         m1 = s1.metadata.Markers.get_item(name)
         assert san_dict(m1._to_dictionary()) == san_dict(m._to_dictionary())
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_save_load_line_segment_marker(self):
         x1, x2, y1, y2 = 1, 9, 4, 7
         color = 'cyan'
@@ -525,6 +534,7 @@ class Test_permanent_markers_io:
         m1 = s1.metadata.Markers.get_item(name)
         assert san_dict(m1._to_dictionary()) == san_dict(m._to_dictionary())
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_save_load_point_marker(self):
         x, y = 9, 8
         color = 'purple'
@@ -541,6 +551,7 @@ class Test_permanent_markers_io:
         m1 = s1.metadata.Markers.get_item(name)
         assert san_dict(m1._to_dictionary()) == san_dict(m._to_dictionary())
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_save_load_rectangle_marker(self):
         x1, x2, y1, y2 = 2, 4, 1, 3
         color = 'yellow'
@@ -558,6 +569,7 @@ class Test_permanent_markers_io:
         m1 = s1.metadata.Markers.get_item(name)
         assert san_dict(m1._to_dictionary()) == san_dict(m._to_dictionary())
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_save_load_text_marker(self):
         x, y = 3, 9.5
         color = 'brown'
@@ -575,6 +587,7 @@ class Test_permanent_markers_io:
         m1 = s1.metadata.Markers.get_item(name)
         assert san_dict(m1._to_dictionary()) == san_dict(m._to_dictionary())
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_save_load_multidim_navigation_marker(self):
         x, y = (1, 2, 3), (5, 6, 7)
         name = 'test point'

--- a/hyperspy/tests/plot/test_plot_markers.py
+++ b/hyperspy/tests/plot/test_plot_markers.py
@@ -131,6 +131,14 @@ class TestMarkers:
         assert m7._get_data_shape() == (2,)
         assert m8._get_data_shape() == (2, 2, 2, 2, 2, 2, 2, 2)
 
+    def test_add_marker_not_plot(self):
+        # This will do nothing, since plot_marker=False and permanent=False
+        # So this test will return a _logger warning
+        s = Signal1D(np.arange(10))
+        m = markers.point(x=5, y=5)
+        s.add_marker(m, plot_marker=False)
+
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_add_marker_signal1d_navigation_dim(self):
         s = Signal1D(np.zeros((3, 50, 50)))
         m0 = markers.point(5, 5)
@@ -141,6 +149,7 @@ class TestMarkers:
             s.add_marker(m1)
         s.add_marker(m2)
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_add_marker_signal2d_navigation_dim(self):
         s = Signal2D(np.zeros((3, 50, 50)))
         m0 = markers.point(5, 5)
@@ -151,6 +160,7 @@ class TestMarkers:
             s.add_marker(m1)
         s.add_marker(m2)
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_add_markers_as_list(self):
         s = Signal1D(np.arange(10))
         marker_list = []
@@ -161,12 +171,20 @@ class TestMarkers:
 
 class Test_permanent_markers:
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_add_permanent_marker(self):
         s = Signal1D(np.arange(10))
         m = markers.point(x=5, y=5)
         s.add_marker(m, permanent=True)
         assert list(s.metadata.Markers)[0][1] == m
 
+    def test_add_permanent_marker_not_plot(self):
+        s = Signal1D(np.arange(10))
+        m = markers.point(x=5, y=5)
+        s.add_marker(m, permanent=True, plot_marker=False)
+        assert list(s.metadata.Markers)[0][1] == m
+
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_remove_permanent_marker_name(self):
         s = Signal1D(np.arange(10))
         m = markers.point(x=5, y=5)
@@ -176,6 +194,7 @@ class Test_permanent_markers:
         del s.metadata.Markers.test
         assert len(list(s.metadata.Markers)) == 0
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_permanent_marker_names(self):
         s = Signal1D(np.arange(10))
         m0 = markers.point(x=5, y=5)
@@ -189,6 +208,7 @@ class Test_permanent_markers:
         assert s.metadata.Markers.test1 == m1
         assert m1.name == 'test1'
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_add_permanent_marker_twice(self):
         s = Signal1D(np.arange(10))
         m = markers.point(x=5, y=5)
@@ -196,6 +216,7 @@ class Test_permanent_markers:
         with pytest.raises(ValueError):
             s.add_marker(m, permanent=True)
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_add_permanent_marker_twice_different_signal(self):
         s0 = Signal1D(np.arange(10))
         s1 = Signal1D(np.arange(10))
@@ -204,6 +225,7 @@ class Test_permanent_markers:
         with pytest.raises(ValueError):
             s1.add_marker(m, permanent=True)
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_add_several_permanent_markers(self):
         s = Signal1D(np.arange(10))
         m_point = markers.point(x=5, y=5)
@@ -226,6 +248,7 @@ class Test_permanent_markers:
         with pytest.raises(ValueError):
             s.add_marker(m_rect, permanent=True)
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_add_markers_as_list(self):
         s = Signal1D(np.arange(10))
         marker_list = []
@@ -234,6 +257,7 @@ class Test_permanent_markers:
         s.add_marker(marker_list, permanent=True)
         assert len(s.metadata.Markers) == 10
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_add_markers_as_list_add_same_twice(self):
         s = Signal1D(np.arange(10))
         marker_list = []
@@ -243,6 +267,7 @@ class Test_permanent_markers:
         with pytest.raises(ValueError):
             s.add_marker(marker_list, permanent=True)
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_add_markers_as_list_add_different_twice(self):
         s = Signal1D(np.arange(10))
         marker_list0 = []
@@ -256,12 +281,14 @@ class Test_permanent_markers:
         s.add_marker(marker_list1, permanent=True)
         assert len(s.metadata.Markers) == 20
     
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_add_permanent_marker_signal2d(self):
         s = Signal2D(np.arange(100).reshape(10, 10))
         m = markers.point(x=5, y=5)
         s.add_marker(m, permanent=True)
         assert list(s.metadata.Markers)[0][1] == m
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     def test_deepcopy_permanent_marker(self):
         x, y, color, name = 2, 9, 'blue', 'test_point'
         s = Signal2D(np.arange(100).reshape(10, 10))

--- a/hyperspy/tests/signal/test_eds_tem.py
+++ b/hyperspy/tests/signal/test_eds_tem.py
@@ -17,6 +17,7 @@
 
 
 import numpy as np
+import pytest
 from matplotlib.testing.decorators import cleanup
 
 from hyperspy.signals import EDSTEMSpectrum

--- a/hyperspy/tests/signal/test_eds_tem.py
+++ b/hyperspy/tests/signal/test_eds_tem.py
@@ -295,6 +295,7 @@ class Test_eds_markers:
                                        weight_percents=[50, 50])
         self.signal = s
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     @cleanup
     def test_plot_auto_add(self):
         s = self.signal
@@ -304,6 +305,7 @@ class Test_eds_markers:
             sorted(s._xray_markers.keys()) ==
             ['Al_Ka', 'Al_Kb', 'Zn_Ka', 'Zn_Kb', 'Zn_La', 'Zn_Lb1'])
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     @cleanup
     def test_manual_add_line(self):
         s = self.signal
@@ -315,6 +317,7 @@ class Test_eds_markers:
         # Check that the line has both a vertical line marker and text marker:
         assert len(s._xray_markers['Zn_La']) == 2
 
+    @pytest.mark.skipif("sys.platform == 'darwin'")
     @cleanup
     def test_manual_remove_element(self):
         s = self.signal

